src/ray/raylet/node_manager.cc

```diff
@@ -2314,7 +2312,7 @@ void NodeManager::HandleDirectCallTaskBlocked(
 void NodeManager::HandleDirectCallTaskUnblocked(
     const std::shared_ptr<WorkerInterface> &worker) {
   if (new_scheduler_enabled_) {
-    if (!worker) {
+    if (!worker || !worker->IsBlocked()) {
```

Comment:
Adding these guards avoids double unblocking a worker (makes unblocking idempotent), which leaks resources permanently.

Reply:
This is because the task may finish before the unblock message arrives right? If so, can you add a comment about that?

Reply:
Done